google/cloud/pubsub_v1/subscriber/_protocol/dispatcher.py

```diff
@@ -164,6 +196,26 @@ def dispatch_callback(self, items: Sequence[RequestItem]) -> None:
         if drop_requests:
             self.drop(drop_requests)
 
+    def _handle_duplicate_request_future(
+        self,
+        exactly_once_delivery_enabled: bool,
+        item: Union[requests.AckRequest, requests.ModAckRequest, requests.NackRequest],
+    ) -> None:
+        _LOGGER.debug(
+            "This is a duplicate %s with the same ack_id: %s.",
+            type(item),
+            item.ack_id,
+        )
+        if exactly_once_delivery_enabled and item.future:
+            item.future.set_exception(ValueError(f"Duplicate ack_id for {type(item)}"))
```

Comment:
Users expect these to be AcknowledgeError. See [sample](https://github.com/googleapis/python-pubsub/blob/79bea329f30dbca20e47d3f31a8b1edbf7603c07/samples/snippets/subscriber.py#L652).

Perhaps we need to add a new DUPLICATE_ACK_ID [AcknowledgeStatus](https://github.com/googleapis/python-pubsub/blob/2fb6e1533192ae81dceee5c71283169a0a85a015/google/cloud/pubsub_v1/subscriber/exceptions.py#L22). Run this by @maheshgattani  as well.

Reply:
Ok, that's good. Thanks for clarifying!